osquery/tables/events/kernel_file_events.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include "osquery/events/kernel.h"
+
+#include <osquery/logger.h>
+
+namespace osquery {
+
+class KernelFileEventSubscriber
+  : public EventSubscriber<KernelEventPublisher> {
+ public:
+  Status init();
+
+  Status Callback(const TypedKernelEventContextRef<osquery_file_event_t> &ec,
+                  const void *user_data);
+};
+
+static osquery_file_event_subscription_t dev_sub = {
+  .actions = (osquery_file_action_t)(OSQUERY_FILE_ACTION_OPEN),
+  .path = "/dev/"
+};
+
+static osquery_file_event_subscription_t var_osquery_sub = {
+  .actions = (osquery_file_action_t)(OSQUERY_FILE_ACTION_OPEN |
+      OSQUERY_FILE_ACTION_CLOSE | OSQUERY_FILE_ACTION_CLOSE_MODIFIED),
+  .path = "/private/var/osquery/"
+};
+
+REGISTER(KernelFileEventSubscriber, "event_subscriber", "kernel_file_events");
+
+Status KernelFileEventSubscriber::init() {
+  auto var_osquery_sc = createSubscriptionContext();
+  var_osquery_sc->event_type = OSQUERY_FILE_EVENT;
+  var_osquery_sc->udata = &var_osquery_sub;
+  subscribe(&KernelFileEventSubscriber::Callback, var_osquery_sc, NULL);
+
+  auto dev_sc = createSubscriptionContext();
+  dev_sc->event_type = OSQUERY_FILE_EVENT;
+  dev_sc->udata = &dev_sub;
+  subscribe(&KernelFileEventSubscriber::Callback, dev_sc, NULL);
```

Comment:
Here I set up subscriptions for two directories.  This seems like it would be better if it were configured via a config file.  Thoughts would be great.